modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -490,10 +490,9 @@ private Model getParent(Model model) {
 
                 // TODO revise below as we've already performed unaliasing so that the isAlias check may be removed
                 Map<String, Object> modelTemplate = (Map<String, Object>) ((List<Object>) models.get("models")).get(0);
-                // Special handling of aliases only applies to Java
                 if (modelTemplate != null && modelTemplate.containsKey("model")) {
                     CodegenModel m = (CodegenModel) modelTemplate.get("model");
-                    if (m.isAlias) {
+                    if (m.isAlias && !ModelUtils.isGenerateAliasAsModel()) {
```

Comment:
@spacether without `&& !ModelUtils.isGenerateAliasAsModel()` it won't work in your case right?

Reply:
That's right

Reply:
FYI. I've filed https://github.com/OpenAPITools/openapi-generator/pull/3951 as `&& !ModelUtils.isGenerateAliasAsModel()` impacts all other generators as well.